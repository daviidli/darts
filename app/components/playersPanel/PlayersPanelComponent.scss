.players {
	height: 15vh;
	text-align: center;

	.player {
		background-color: rgba(0, 0, 0, 0.4);
		height: 100%;
		margin: 0 1em;
		max-width: 25em;

		&.active {
			background-color: rgba(249, 198, 96, 0.4);
		}
	}

	.playerTotal {
		font-weight: 700;
	}
}
