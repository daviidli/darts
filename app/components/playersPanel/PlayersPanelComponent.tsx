import React from 'react';
import { Grid, Typography } from '@material-ui/core';
import clsx from 'clsx';
import { playerType } from '../../reducers/types';
import styles from './PlayersPanelComponent.scss';

export type Props = {
	totals: any[];
	players: playerType[];
	currentPlayer: number;
};

const PlayersPanelComponent = (props: Props) => {
	const { totals, players, currentPlayer } = props;

	return (
		<Grid
			container
			direction="row"
			justify="center"
			alignItems="center"
			className={styles.players}
		>
			{players.map((player, i) => (
				<Grid
					item
					xs
					container
					direction="column"
					justify="center"
					alignItems="center"
					key={player.id}
					className={clsx(styles.player, {
						[styles.active]: i === currentPlayer
					})}
				>
					<Grid item>
						<Typography variant="h2" className={styles.playerTotal}>
							{totals[i]}
						</Typography>
					</Grid>
					<Grid item>{player.name}</Grid>
				</Grid>
			))}
		</Grid>
	);
};

export default PlayersPanelComponent;
