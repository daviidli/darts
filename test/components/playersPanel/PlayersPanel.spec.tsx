import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import PlayersPanelComponent, {
	Props
} from '../../../app/components/playersPanel/PlayersPanelComponent';

Enzyme.configure({ adapter: new Adapter() });

const initialProps: Props = {
	totals: [0],
	players: [{ id: '1', name: 'player1' }],
	currentPlayer: 0
};

const setup = (props: Props = initialProps) => {
	const wrapper = shallow(<PlayersPanelComponent {...props} />);
	return { wrapper };
};

describe('PlayersPanel component', () => {
	it('should match snapshot', () => {
		const { wrapper } = setup();
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it('should match snapshot with 2 players', () => {
		const props = {
			totals: [100, 200],
			players: [
				{ id: '1', name: 'player1' },
				{ id: '2', name: 'player2' }
			],
			currentPlayer: 1
		};
		const { wrapper } = setup(props);
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
