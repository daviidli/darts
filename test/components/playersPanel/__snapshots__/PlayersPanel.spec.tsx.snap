// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayersPanel component should match snapshot 1`] = `
<WithStyles(ForwardRef(Grid))
  alignItems="center"
  className="players"
  container={true}
  direction="row"
  justify="center"
>
  <WithStyles(ForwardRef(Grid))
    alignItems="center"
    className="player active"
    container={true}
    direction="column"
    item={true}
    justify="center"
    key="1"
    xs={true}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <WithStyles(ForwardRef(Typography))
        className="playerTotal"
        variant="h2"
      >
        0
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      player1
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`PlayersPanel component should match snapshot with 2 players 1`] = `
<WithStyles(ForwardRef(Grid))
  alignItems="center"
  className="players"
  container={true}
  direction="row"
  justify="center"
>
  <WithStyles(ForwardRef(Grid))
    alignItems="center"
    className="player"
    container={true}
    direction="column"
    item={true}
    justify="center"
    key="1"
    xs={true}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <WithStyles(ForwardRef(Typography))
        className="playerTotal"
        variant="h2"
      >
        100
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      player1
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    alignItems="center"
    className="player active"
    container={true}
    direction="column"
    item={true}
    justify="center"
    key="2"
    xs={true}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <WithStyles(ForwardRef(Typography))
        className="playerTotal"
        variant="h2"
      >
        200
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      player2
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
